packages/jsii-rosetta/lib/languages/java.ts

```diff
@@ -0,0 +1,629 @@
+      this.renderTypeNode(node.type, renderer, 'void'));
+  }
+
+  public parameterDeclaration(node: ts.ParameterDeclaration, renderer: JavaRenderer): OTree {
+    return new OTree([
+      this.renderTypeNode(node.type, renderer),
+      ' ',
+      renderer.convert(node.name),
+    ]);
+  }
+
+  public block(node: ts.Block, renderer: JavaRenderer): OTree {
+    return this.renderBlock(renderer.convertAll(node.statements));
+  }
+
+  public variableDeclaration(node: ts.VariableDeclaration, renderer: JavaRenderer): OTree {
+    const type = (node.type && renderer.typeOfType(node.type))
+        || (node.initializer && renderer.typeOfExpression(node.initializer));
+
+    const renderedType = type ? this.renderType(type, renderer, 'Object') : 'Object';
+
+    return new OTree(
+      [
+        renderedType,
+        ' ',
+        renderer.convert(node.name),
+        ' = ',
+        renderer.convert(node.initializer),
+        ';'
+      ],
+      [],
+      {
+        canBreakLine: true,
+      }
+    );
+  }
+
+  public expressionStatement(node: ts.ExpressionStatement, renderer: JavaRenderer): OTree {
+    const inner = renderer.convert(node.expression);
+    return inner.isEmpty
+      ? inner
+      : new OTree([inner, ';'], [], { canBreakLine: true });
+  }
+
+  public ifStatement(node: ts.IfStatement, renderer: JavaRenderer): OTree {
+    const ifStmt = new OTree(
+      [
+        'if (',
+        renderer.convert(node.expression),
+        ') ',
+      ],
+      [
+        renderer.convert(node.thenStatement),
+      ],
+      {
+        canBreakLine: true,
+      },
+    );
+    const elseStmt = node.elseStatement
+      ? new OTree(['else '], [renderer.convert(node.elseStatement)], { canBreakLine: true })
+      : undefined;
+
+    return elseStmt
+      ? new OTree(
+        [],
+        [ifStmt, elseStmt],
+        {
+          separator: ' ',
+          canBreakLine: true,
+        },
+      )
+      : ifStmt;
+  }
+
+  public forOfStatement(node: ts.ForOfStatement, renderer: JavaRenderer): OTree {
+    // This is what a "for (const x of ...)" looks like in the AST
+    let variableName = '???';
+
+    matchAst(node.initializer,
+      nodeOfType(ts.SyntaxKind.VariableDeclarationList,
+        nodeOfType('var', ts.SyntaxKind.VariableDeclaration)),
+      bindings => {
+        variableName = renderer.textOf(bindings.var.name);
+      });
+
+    return new OTree(
+      [
+        'for (Object ',
+        variableName,
+        ' : ',
+        renderer.convert(node.expression),
+        ') ',
+      ],
+      [
+        renderer.convert(node.statement),
+      ],
+      {
+        canBreakLine: true,
+      },
+    );
+  }
+
+  public printStatement(args: ts.NodeArray<ts.Expression>, renderer: JavaRenderer) {
+    return new OTree(
+      [
+        'System.out.println(',
+        args.length === 1
+          ? renderer.convert(args[0])
+          : new OTree([], ['""', ...renderer.convertAll(args)], { separator: ' + ' }),
```

Comment:
See Chime comment. Even though this may be semantically correct, I don't think this looks very nice to a user reading code samples. Can we simplify? If we have to sacrifice correctness in edge cases, that's okay, this doesn't have to be executed anyway.

Reply:
Changed.